event.go

```diff
@@ -223,11 +257,82 @@ func (e *Event) Equals(ev *Event) bool {
 	return true
 }
 
-// Len calculates the length of the string representation of event. Note that
-// this will return the true length (even if longer than what IRC supports),
-// which may be useful if you are trying to check and see if a message is
-// too long, to trim it down yourself.
+// split will split a potentially large event that is larger than what the server
+// supports, into multiple events. split will ignore events that cannot be split, and
+// if the event isn't longer than what the server supports, it will just return an array
+// with 1 entry, the original event.
+func (e *Event) split(maxLength int) []*Event {
+	if len(e.Params) < 1 || (e.Command != PRIVMSG && e.Command != NOTICE) {
+		return []*Event{e}
+	}
+
+	// Exclude source, even if it does exist, because the server will likely ignore the
+	// sent source anyway.
+	event := e.Copy()
+	event.Source = nil
+
+	if event.LenOpts(false) < maxLength {
+		return []*Event{e}
+	}
+
+	results := []*Event{}
+
+	// Will force the length check to include " :". This will allow us to get the length
+	// of the commands and necessary prefixes.
+	text := event.Last()
+	event.Params[len(event.Params)-1] = ""
+	cmdLen := event.LenOpts(false)
+
+	var ok bool
+	var ctcp *CTCPEvent
+	if ok, ctcp = e.IsCTCP(); ok {
+		if len(text) == 0 {
+			return []*Event{e}
+		}
+
+		text = ctcp.Text
+
+		// ctcpDelim's at start and end, and space between command and trailing text.
+		maxLength -= len(ctcp.Command) + 4
+	}
+
+	// TODO: colors? use last color at start of split? make sure it's POST-color gen?
+
+	// If the command itself is longer than the limit, there is a problem. PRIVMSG should
+	// be 1->1 per RFC. Just return the original message and let it be the user of the
+	// libraries problem.
+	if cmdLen > maxLength {
+		return []*Event{e}
+	}
+
+	// Split the text into correctly size segments, and make the necessary number of
+	// events that duplicate the original event.
+	for _, split := range splitAtWord(text, maxLength-cmdLen) {
```

Comment:
I don't think there is a need to split things other than PRIVMSG/NOTICE this way. Mostly because there are so few things that can actually be split on the client side. As far as I've gathered, it's really just PRIVMSG, NOTICE, and JOIN. I see weechat has support for splitting ISUPPORT, but I've never seen client->server ISUPPORT. There may be more, but IMO, users should use `girc.Commands`, so that's where command-specific splitting should happen.

Furthermore, as we already have JOIN splitting supported (and it was just updated in this PR), I don't think there is a need to overengineer the splitting to support extensibility. It's not like the RFC protocol is ever going to change (maybe a better defined RFC, but not updates). IRCv3 has updates to the protocol, however it wouldn't make sense for IRCv3 to go backwards, and require this kind of splitting for IRCv3 native features.